forge-kernel/src/VoxelIoU.cpp

```
@@ -0,0 +1,286 @@
+    long inA = 0, inB = 0, both = 0, either = 0, errs = 0;
+    for (int i = 0; i < gridN; ++i) {
+        const double x = lo[0] + (i + 0.5) * step[0];
+        for (int j = 0; j < gridN; ++j) {
+            const double y = lo[1] + (j + 0.5) * step[1];
+            for (int k = 0; k < gridN; ++k) {
+                const gp_Pnt p(x, y, lo[2] + (k + 0.5) * step[2]);
+                bool a = false, b = false;
+                try {
+                    ca.Perform(p, tol);
+                    a = (ca.State() == TopAbs_IN || ca.State() == TopAbs_ON);
+                } catch (...) { ++errs; }
```

Clang (14.0.6)

[note] +4, including nesting penalty of 3, nesting level increased to 4

(clang)